### PR TITLE
[MRG] Optimize read_source_estimate function

### PR DESCRIPTION
**Pull request by wmvanvliet**
I'm having a weird use case for which I need to load in 2413 source estimate objects. While waiting for it to load, I looked at the source code and found a simple way to make the loading roughly twice as fast. Reading an stc is probably almost never a bottleneck, but hey, it's a small change *shrug*.

Seeking to the end of the file can be a costly operation on some file systems, especially when reading from a network mount. Since all bytes of the file are read during this function anyway, it makes more sense to
just read everything into a buffer instead of doing file seeks.

### CODE REVIEW

File: mne/source_estimate.py [hunk 1]
```diff
@@ -33,41 +33,42 @@
...
 def _read_stc(filename):
     """Aux Function."""
-    fid = open(filename, 'rb')
+    with open(filename, 'rb') as fid:
+        buf = fid.read()
 
     stc = dict()
-
-    fid.seek(0, 2)  # go to end of file
-    file_length = fid.tell()
```
Comment (agramfort) [inline mne/source_estimate.py]: this was costly?

File: mne/source_estimate.py [hunk 2]
```diff
@@ -33,41 +33,42 @@
...
 
     # read the source vector
-    stc['vertices'] = np.fromfile(fid, dtype=">u4", count=vertices_n)
+    stc['vertices'] = np.frombuffer(buf, dtype=">u4", count=vertices_n, offset=offset)
+    offset += np.dtype('>u4').itemsize * vertices_n
 
     # read the number of timepts
-    data_n = int(np.fromfile(fid, dtype=">u4", count=1))
+    data_n = int(np.frombuffer(buf, dtype=">u4", count=1, offset=offset))
+    offset += np.dtype('>u4').itemsize
```
Comment (jaeilepp) [inline mne/source_estimate.py]: It's a bit silly to do this for every type. I would do ``n_bytes = 4`` and use that everywhere.

File: mne/source_estimate.py [hunk 3]
```diff
@@ -33,41 +33,42 @@
...
 
     if (vertices_n and  # vertices_n can be 0 (empty stc)
-            ((file_length / 4 - 4 - vertices_n) % (data_n * vertices_n)) != 0):
+            ((len(buf) / 4 - 4 - vertices_n) % (data_n * vertices_n)) != 0):
```
Comment (jaeilepp) [inline mne/source_estimate.py]: Here the 4's should probably be ``n_bytes`` too.

File: mne/source_estimate.py (hunk 1 above)
Comment (wmvanvliet) [inline mne/source_estimate.py]: yes, it was reading in the entire file. It probably depends on which filesystem you use. I'm using a mounted NFS share.

File: mne/source_estimate.py (hunk 3 above)
Comment (Eric89GXL) [inline mne/source_estimate.py]: and for integer arithmetic we should really use `//` (not introduced by you but we should probably fix it while you're digging around in there)

**Comment by agramfort** [comment]
travis complains

**Comment by agramfort** [comment]
thx @wmvanvliet

**Pull request merged**